app/experimenter/experiments/tests/api/v5/test_mutation.py

```diff
@@ -160,3 +193,65 @@ def test_update_experiment_error(self):
             {"name": ["Ensure this field has no more than 255 characters."]},
         )
         self.assertEqual(result["status"], 200)
+
+    def test_update_experiment_branches_with_feature_config(self):
+        user_email = "user@example.com"
+        feature = NimbusFeatureConfigFactory()
+        exp = NimbusExperimentFactory.create_with_status(NimbusExperiment.Status.DRAFT)
+        control_branch = {"name": "control", "description": "a control", "ratio": 1}
+        treatment_branches = [{"name": "treatment1", "description": "desc1", "ratio": 1}]
+        response = self.query(
+            UPDATE_EXPERIMENT_BRANCHES_MUTATION,
+            variables={
+                "input": {
+                    "nimbusExperimentId": exp.id,
+                    "featureConfigId": feature.id,
+                    "clientMutationId": "randomid",
+                    "controlBranch": control_branch,
+                    "treatmentBranches": treatment_branches,
+                }
+            },
+            headers={settings.OPENIDC_EMAIL_HEADER: user_email},
+        )
+        self.assertEqual(response.status_code, 200)
+        content = json.loads(response.content)
+        result = content["data"]["updateExperimentBranches"]
+        self.assertEqual(
+            result["nimbusExperiment"],
+            {
+                "id": str(exp.id),
+                "featureConfig": {"name": feature.name},
+                "status": exp.status.upper(),
+                "name": exp.name,
+                "slug": exp.slug,
+                "controlBranch": control_branch,
+                "treatmentBranches": treatment_branches,
+            },
```

Comment:
What does that look like?

Reply:
So we're passing in two branches, one gets set to control, so roughly like

```py
self.assertEqual(experiment.feature_config, feature)
self.assertEqual(experiment.branches.count(), 2)
self.assertEqual(experiment.control_branch.name, control_branch['name'])
treatment_branch = experiment.branches.exclude(id=experiment.control_branch.id).get()
self.assertEqual(treatment_branch.name, treatment_branches[0]['name'])
```

Don't need to assert all the details of all the fields since that's covered by serializer tests, just that they got set.